apps/picsa-tools/option-tool/src/app/pages/home/home.component.ts

```diff
@@ -29,11 +33,30 @@ export class HomeComponent {
   public closeDialog(){
     this.matStepperOpen = false
   }
-  onDataTransfer(data: IOptionData) {
+  onDataTransfer(data:{data:IOptionData,index:number}) {
     console.log('Received data from editor:', data);
     // close the child component
-    this.dataSource = [...this.dataSource, data];
+    if(data.index > -1){
+    this.dataSource[data.index]=data.data
+    //invoke change detaction
```

Comment:
nice catch, yes change detection isn't triggered when nested inside an object or array.

One option like you have used is to recreate the entire array, the other would be to manually call angular's change detection method as outlined in https://angular.io/api/core/ChangeDetectorRef

For now I think you are fine with the method you have chosen, so more just sharing in case of interest

Reply:
Alright thank you